docs/features/analysis.md

```diff
@@ -425,6 +425,48 @@ spec:
           valueFrom:
             podTemplateHashValue: Latest
 ```
+Analysis arguments also support valueFrom for reading metadata fields and passing them as arguments to AnalysisTemplate.
+An example would be to reference metadata labels like env and region and passing them along to AnalysisTemplate.
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: guestbook
+  env: test
+  region: us-west-2
+spec:
+...
+  strategy:
+    canary:
+      analysis:
+        templates:
+        - templateName: args-example
+        args:
+        # required value 
+        - name: service-name
+          value: guestbook-svc.default.svc.cluster.local
+        # override default value
+        - name: api-url
+          value: http://other-api
+        # pod template hash from the stable ReplicaSet
+        - name: stable-hash
+          valueFrom:
+            podTemplateHashValue: Stable
+        # pod template hash from the latest ReplicaSet
+        - name: latest-hash
+          valueFrom:
+            podTemplateHashValue: Latest
```

Comment:
To simplify the documentation, I think we can omit these entirely since they do not highlight the feature.

Reply:
ok , good idea , i have removed this part as per your suggestion.